### DISCUSSION TITLE
Trouble remote controlling kitty from nvim

### DISCUSSION THREAD

**Discussion by samEngineering** [kovidgoyal/kitty#6935] [General]
After reading the docs about `allow_remote_control` and `listen_on`, and reading [this discussion](https://github.com/kovidgoyal/kitty/commit/b348195d22f75d47d57e9865ec44b11e046c350b), I am still a little short on understanding.
My goal is to run `kitty @ set-spacing padding=0` from within `nvim` so that the margins look nice for the tabline while running `nvim` (and then reverting after `nvim` is closed. In my `kitty` config, I have `allow_remote_control   yes`.

on the `nvim` side, I have
`os.execute("kitty @ set-spacing padding=0")`
in my `init.lua` but no result.

I also tried
`os.execute("kitty @ set-spacing padding=0 && kill -s SIGUSR1 <pid>")`,
where `<pid>` is the pid of the current kitty instance. No dice there either. I tried each of the above commands with `kitten` in place of `kitty` as well. Executing the command `kitty @ set-spacing padding=0` on the command line in the terminal (outside of `nvim`) works as expected.

I have verified that the `os.execute` call is working by putting
`os.execute("echo 'hello, world!' > file.txt")`,
in the `init.lua` and verifying the existence and contents of `file.txt`. I am hoping someone with more knowledge of remote control can lend some tips ^_^

**Comment by kovidgoyal** [comment]
neovim does not connect processes launched by os.execute() to the tty
device. So either setup a socket for remote control, or use vim rather
than neovim or run kitty from master and use a watcher witch listens for
the setting of user variables and configure neovim to set a user
variable of startup and delete it on shutdown.

**Reply by samEngineering** [reply]
So helpful, thank you!
I'll read up on sockets and work on these suggestions.
